lib/json0.js

```diff
@@ -663,6 +663,61 @@ json.transformComponent = function(dest, c, otherC, type) {
   return dest;
 };
 
+json.transformPresence = function(presence, op, isOwnOp) {
+  if (!presence || !isArray(presence.p)) return null;
+  if (!op) return presence;
+
+  presence = clone(presence);
+  op = clone(op);
+
+  // Deletions should be treated as insertions: transforming
+  // by them should result in a no-op (null presence), so
+  // let's just pretend that deletions are oi to get the
+  // desired transform behaviour
+  for (var i = 0; i < op.length; i++) {
+    const component = op[i];
+    if ('od' in component) {
+      component.oi = component.od;
+      delete component.od;
+    }
+
+    if ('ld' in component) {
+      component.oi = component.ld;
+      delete component.ld;
+    }
+
+    // Handle text0 ops using the subtype
+    if ('si' in component || 'sd' in component) {
+      convertFromText(component);
+    }
+  }
+
+  // Create a fake op so we can transform the presence path using
+  // existing machinery
+  var transformed = [{p: presence.p, oi: ''}];
+
+  for (var i = 0; i < op.length; i++) {
+    var component = op[i];
+    // Set side as 'right' because we always want the op to win ties, since
+    // our transformed "op" isn't really an op
```

Comment:
Would be worth mentioning that for json0, this is just to handle subpath changes as a result of list move, insert, delete.

Reply:
Done